### ISSUE TITLE
DataFusion weekly project plan (Andrew Lamb) - June 3, 2024

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#10779]
Follow on to https://github.com/apache/datafusion/issues/10699

My (personal) North ⭐ : 1000 projects are built using DataFusion 📈 

**It would be great for other contributors to DataFusion who plan non trivial work could try to make them visible somehow as well** 🙏  (feel free to copy / modify the format)

## External DataFusion 
* DataFusion is getting lots of hilights: @philippemnoel  Search and Anaytics  with Postgres on the GeekNarrator: https://www.youtube.com/watch?v=aODapXAuMNI. I haven't watched this yet but hope to this week
* I found out that CMU's Advanced DB class has been using DataFusion this semeseter: https://twitter.com/andrewlamb1111/status/1796623935477354813
* @andygrove has been running [comet community meetings](https://lists.apache.org/thread/vn55tlc1rpx5lj1pmfqlo65mw0q0tq6m) like a 👑 

## Feature Highlights
* Merged example for using an external index with parquet: https://github.com/apache/datafusion/issues/10546
*`ParquetExec` is getting better and better: https://github.com/apache/datafusion/pull/10636, https://github.com/apache/datafusion/pull/10716 (thanks @ jeffreyssmith2nd)
* @Blizzara  and @waynexia  and @jonahgao  are improving substrait support (e.g https://github.com/apache/datafusion/pull/10653)
* Listing table now pushes down predicates to directories (thanks @houqp  https://github.com/apache/datafusion/pull/10693)
* We almost have upgraded sqlparser (thanks @tisonkun !): https://github.com/apache/datafusion/pull/10392
* Parquet statistics tests and functionality are getting quite complete out thanks to @xinlifoobar, @Lordworms and @Weijun-H -- https://github.com/apache/datafusion/pull/10775 https://github.com/apache/datafusion/pull/10711, etc
* @goldmedal began making the "query from file urls" DynamicScemaProvider available outside datafusion-cli: https://github.com/apache/datafusion/pull/10745
* Started a discussion about bringing sqlparser-rs into DataFusion project: https://github.com/sqlparser-rs/sqlparser-rs/issues/1294

## My (personal) plans for this week
TLDR is I plan to complete the API for pushing down row selections into ParquetExec
- [ ] Keep hacking on https://github.com/apache/datafusion/issues/10453
- [ ] https://github.com/apache/arrow-datafusion/issues/9929
- [ ] Prepare for SIGMOD next week. Will record the talk https://github.com/apache/datafusion/issues/10480
- [ ] https://github.com/apache/datafusion/issues/10481
- [ ] Help  @tustvold  complete https://github.com/apache/arrow-rs/issues/5736 (by reviewing)

## Project Queue (list of future projects)
- [ ] (this weekend likely) https://github.com/apache/datafusion/issues/9873 copies (last part of  https://github.com/apache/arrow-datafusion/issues/9637)
- [ ] In general, help https://github.com/apache/arrow-rs/issues/5374 more actively as it seems to be stuck

## Projects I plan to help actively help review / plan
- [ ] https://github.com/apache/arrow-datafusion/issues/9494

# Algorithm for (my) prioritizing PR reviews
Note there are many [committers](https://projects.apache.org/committee.html?datafusion) who can and do review and merge PRs, so this is not the priorities of the project as a whole, just the approximate algorithm I use to prioritize my own time.

Priority:
1. Bug fixes (where something is just incorrect), especially regressions (where it used to work and now does not)
2. Improvements directly related to features needed for InfluxDB (my employer)
3. Documentation and test improvements (I view these as very strategically important)
4. PRs that I think are strategically important
5. Other new features / additions to functionality (note this is the lowest)

The top strategically important projects in my head are:

* Anything that makes it easier to use DataFusion as a user (docs, examples, better APIs, etc), including [Getting started guide for new users (who want to use DataFusion in their project) ](https://github.com/apache/arrow-datafusion/issues/7014) 
* Anything that improves DataFusion's quality such as bug fixes, and improved / expanded tests, etc
* Making function packages modular and easier to mix/match 
* Improved performance (with benchmarks)

Thus, if you are interested in contributing to DataFusion and are interested in a fast turn around time I would recommend looking into bug fixes / test improvements / documentation or the projects named above. 

If you propose adding new functionality, especially if the PR is large/complex and not connected to a wider need, the review cycle will likely be longer. You can make it a shorter cycle by looking at the comments on other recent PRs and following the same model (e.g. ensure there are tests in sqllogictest for example, the CI passes, includes documentation, etc)

## Background
The idea of this ticket is make my plans for DataFusion visible, largely for my own personal organizational needs, but also to:
1. Communicate / coordinate in the community 
2. Help provide an interesting summary of what is happening in DataFusion this week

**Comment by alamb** [comment]
https://github.com/apache/datafusion/issues/10869 next week
